src/BaseTransformation.cpp

```diff
@@ -0,0 +1,33 @@
+#include "../include/BaseTransformation.h"
+
+using clang::IfStmt;
+using llvm::isa, llvm::cast;
+
+BasicVisitor::BasicVisitor(Rewriter * rewriter) : rewriter(rewriter) {}
+
+bool BasicVisitor::VisitStmt(Stmt *s) {
```

Comment:
This is isn't `BaseVisitor`. We can implement `AddCommentTransformation`, which is inheritance from base and put comments in defined places.